src/OSPSuite.Core/Domain/Services/ContainerMergeTask.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System.Linq;
+using OSPSuite.Utility.Extensions;
+
+namespace OSPSuite.Core.Domain.Services
+{
+   public interface IContainerMergeTask
+   {
+      void MergeContainers(IContainer targetContainer, IContainer containerToMerge);
+      void AddOrMergeContainer(IContainer parentContainer, IContainer containerToMerge);
+
+      /// <summary>
+      ///    If the entity exists, it will be removed and replace by the new one, otherwise simply added
+      /// </summary>
+      void AddOrReplaceInContainer(IContainer container, IEntity entityToAddOrReplace);
+   }
+
+   public class ContainerMergeTask : IContainerMergeTask
```

Comment:
I did not add any test for this because this is already fully tested from the behavior in SpastialStrucutreMerger and also the new tests for neighborhood